Fix an issue with authored state.

scytacki opened the pull request:
LARA was updated at some point to send the empty object `{}` as the initial authored state.
I think previously the state was null or the empty string.
This broke the table interactive since it needed a default authored state and was using
`authoredState || DEAULT_AUTHORED_STATE` to load the default state.
Initially the authoredState being `{}` meant it would be truthy and the default state would
never be loaded.

scytacki committed:
Fix an issue with authored state.

LARA was updated at some point to send the empty object `{}` as the initial authored state.
I think previously the state was null or the empty string.
This broke the table interactive since it needed a default authored state and was using
`authoredState || DEAULT_AUTHORED_STATE` to load the default state.
Initially the authoredState being `{}` meant it it would be truthy and the default state would
never be loaded.

scytacki committed:
try to fix issue with travis

$ travis_retry pip install awscli --upgrade --user
Collecting awscli
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#snimissingwarning.

kswenson commented on js/components/app.js:
```diff
@@ -46,6 +46,10 @@ const DEFAULT_AUTHORED_STATE = {
 };
 
+function validAuthoredState(state) {
+  return state && state.columns
```
Wouldn't it be something like this instead:
```suggestion
  return state && state.columns ? state : undefined
```

scytacki replied:
Yes you are right. Thanks. :)

scytacki committed:
need to handle empty string from portal-report

For some reason the portal-report sends an empty string for the authored state
instead of an empty object or null. So this code should handle all cases now.

scytacki committed:
fixed issue with returning actual authored state

scytacki commented:
@kswenson since you already looked at this I'm re-assigning it to you. But if you don't want it, I can give it PIotr.

This is a legacy interactive that I like to use to demonstrate state saving and authoring. 
It is used in several activities still.

kswenson commented on js/components/app.js:
```diff
@@ -46,6 +46,25 @@ const DEFAULT_AUTHORED_STATE = {
+    if (value === "") {
+      return null;
+    } else {
+      return JSON.parse(value);
```
Do you need to worry about the case of a non-parseable string? If so, you could wrap in a try/catch block.

scytacki replied:
Yeah it would be good to do that, but I'll punt it for now though. If there is an error in the string we should see the exception in the console and hopefully can track that down.  Ideally we'd put up some user dialog so if this happens a user knows something is awry.

kswenson commented on package.json:
```diff
@@ -4,7 +4,9 @@
+    "start": "webpack-dev-server",
+    "start:secure": "webpack-dev-server --https --inline --hot --content-base dist/ --cert ~/.localhost-ssl/localhost.pem --key ~/.localhost-ssl/localhost.key"
   },
   "author": "Piotr Janik <janikpiotrek@gmail.com>",
```
Not relevant to this PR, but I've often wondered whether we (should) have a policy on how to handle these `author` fields. Should they be individual developers or should they always be `developer@concord.org`, for instance?

scytacki replied:
Yeah I'm never sure either. It is kind of nice to have attribution for the people that have worked on the project. But the git history and github repo info also provides that. I guess from my point of view, in most of these cases no one outside of CC will look at this code so it doesn't much matter. But if you want to propose a policy of using developer@concord.org on the `#dev` channel I'd be fine with that.

kswenson reviewed (approved):
👍 LGTM -- just a couple of questions that don't (necessarily) require changes
